src/viam/ur/module/ur_arm_state.hpp

```diff
@@ -285,7 +287,15 @@ class URArm::state_ {
     // URArm misusing it.
     struct move_request {
        public:
-        explicit move_request(std::vector<trajectory_sample_point>&& samples, std::ofstream arm_joint_positions_stream);
+        using move_command_data = std::variant<std::vector<trajectory_sample_point>, std::optional<pose_sample>>;
+
+        explicit move_request(move_command_data&& move_command, std::ofstream arm_joint_positions_stream);
+
+        explicit move_request(std::vector<trajectory_sample_point>&& tsps, std::ofstream arm_joint_positions_stream)
+            : move_request(move_command_data{std::move(tsps)}, std::move(arm_joint_positions_stream)) {}
+
+        explicit move_request(pose_sample ps, std::ofstream arm_joint_positions_stream)
```

Comment:
A quibble: if we don't log joint positions from a move to pose, should this ctor take an `arm_join_positions_stream`? Or, _should_ we be logging something during a move to pose, like we do with joint positions for a `move_through_joint_positions? In which case, the field and parameter names probably need a fixup (as does the controlled state machine arm for the pose case).